arelle/typing.py

```diff
@@ -1,5 +1,6 @@
 """Type hints for Arelle."""
 from __future__ import annotations
+from typing import Any
 from collections.abc import Callable
 
-TypeGetText = Callable[[str], str]
+TypeGetText = Callable[[str], Any]
```

Comment:
Are you sure about this? Doesn't the translation feature always return a translated string?

Reply:
> Are you sure about this? Doesn't the translation feature always return a translated string?

Not sure, I am digging into this, what I know for sure it caused me some trouble when returning arabic.

Reply:
What do you get if you run `print(type("Some text in arabic"))`?

Reply:
string, but it will depend on where gettext is pulling the lines from

Reply:
https://docs.python.org/3.9/library/gettext.html#gettext.gettext 
Not sure which functions are the warning refers to.

Reply:
You can use this trick to find out what the type is. What I normally do is print the type and then run through a test suite with `-s` at the end. You should get the same results if you parse a stand-alone file with the command line.

Reply:
I think this more about the po files (translation files) and how they are encoded. Normally for the fallback language (English) it will return the English text as is _("Text here"), so it will always be a string, when you start pulling text from the po files you might get a different type.

Reply:
Return type `Any` confirmed, steps:
```python
# add these lines befor by mypy run (say starting line Cntrl.py:287)
 reveal_type(_("user name if needed (with password) for web file retrieval"))
 self.setUiLanguage('ar-EG', True)
 reveal_type(_("user name if needed (with password) for web file retrieval"))
```
Result:
```bash
arelle/Cntlr.py:287: note: Revealed type is "Any"
arelle/Cntlr.py:289: note: Revealed type is "Any"
```
`print(type(<thing>))` is tricky, because `str` is `Any`